app/src/main/java/com/example/returnpals/ScheduleReturn.kt

```diff
@@ -29,130 +34,161 @@ import java.time.LocalDate
-                pickUpAddress = state.value.address!!,
-                nextButton = { step.intValue++ },
-                backButton = { step.intValue-- },
-                promoButton = {}
-            )
-        7 -> thanks.drawThankYouUI(
-                dashBoardButton = { /* TODO: navigate to dashboard */ },
-            )
-    }
+    ScheduleReturnContent(step.value, pickupVM,
+        onClickNext = { step.value++; Log.println(Log.INFO, "onClickNext", step.value.toString()) },
+        onClickBack = { step.value--; Log.println(Log.INFO, "onClickBack", step.value.toString()) },
+        onSubmit = { goto(navController, MenuRoutes.HomeDash) }
+    )
 
 }
 
-/////////////////////////////////////////////////////////////////////////////
-// PRIVATE API
-////////////////////
-// other files need not be aware of or concerned with everything below here
+open class PickupViewModel(
+    pickup: PickupInfo,
+) : ViewModel() {
 
-private class ScheduleReturnViewModel(init: PickupInfo) : androidx.lifecycle.ViewModel() {
+    private val _idManager: IdManager = IdManager()
+    private val _data: MutableLiveData<PickupInfo> = MutableLiveData(pickup)
+    val data: LiveData<PickupInfo> = _data
+    private val _minDate: LocalDate = LocalDate.now().minusDays(1)
+    private val _maxDate: LocalDate = LocalDate.now().plusMonths(2)
 
-    private val idManager: IdManager = IdManager()
-    private val _pickup: MutableStateFlow<PickupInfo>
-    val pickup: StateFlow<PickupInfo>
+    // not sure why _data.value is nullable...
 
-    init {
-        _pickup = MutableStateFlow(init)
-        pickup = _pickup.asStateFlow()
-        // _pickup.value and pickup.value should NOT be null after init
+    fun isValidDate(value: LocalDate): Boolean {
+        return value > _minDate && value < _maxDate
     }
 
     fun onChangeDate(value: LocalDate) {
-        _pickup.update {
-            it.date = value
-            it
-        }
+        if (isValidDate(value)) _data.value!!.date = value
     }
 
     fun onChangeAddress(value: Address) {
-        _pickup.update {
-            it.address = value
-            it
-        }
+        _data.value!!.address = value
     }
 
-    fun onChangePickupMethod(value: PickupMethod) {
-        _pickup.update {
-            it.method = value
-            it
-        }
+    fun onChangeMethod(value: PickupMethod) {
+        _data.value!!.method = value
     }
 
     fun onAddLabel(value: PackageInfo) {
-        _pickup.update {
-            val label = value.copy(id=idManager.allot())
-            it.packages[label.id] = label
-            it
-        }
+        val id = _idManager.allot()
+        _data.value!!.packages[id] = value.copy(id=id)
     }
 
     fun onRemoveLabel(id: Long) {
-        _pickup.update {
-            it.packages.remove(id)
-            idManager.free(id)
-            it
-        }
+        _data.value!!.packages.remove(id)
+        _idManager.free(id)
+    }
+
+    fun onChangePlan(value: PricingPlan) {
```

Comment:
These are looking great. I like the naming makes everything clear. Should we still add our function comment?

Reply:
Schedule Return ViewModel is still in development, it'll be implemented, commented, and tested a by the end of this sprint.